app/uk/gov/hmrc/agentclientrelationships/connectors/UsersGroupsSearchConnector.scala

```diff
@@ -49,6 +63,17 @@ class UsersGroupsSearchConnector @Inject()(
     extends HttpAPIMonitor {
   override val kenshooRegistry: MetricRegistry = metrics.defaultRegistry
 
+  def getFirstGroupAdminUser(groupId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[UserDetails] = {
+    val url = new URL(baseUrl, s"/users-groups-search/groups/$groupId/users")
+    monitor(s"ConsumedAPI-UGS-getGroupUsers-GET") {
+      httpGet
+        .GET[Seq[UserDetails]](url.toString)
+        .map(_.find(_.credentialRole.exists(_ == "Admin")).getOrElse(throw AdminNotFound("NO_ADMIN_USER")))
```

Comment:
not sure if this creates overhead on the UGS service as we are retrieving all the users from the group? 
I mean think if the group is too big?

Reply:
Well, I'm just implementing the algorithm I was given.  You are correct that this might be a very large list, but at the moment we are making one call per user and that hammers their system.  I have tried to mitigate this by only creating a small case class with what we need, and I understand the number of users in a group is generally < 100.  We will need to check this though - it's a good call!

Reply:
Also, note that this will (I think) find the first admin user, and it should (I hope) short-circuit when it finds one.  The worst-case scenario is when the admin user is the *last* user in the list - I wonder how common that would be?

Reply:
it does not matter even if the admin exists in the front or in the last of the list, because this processing is done on our end. 
My concern is if the group contains 100,000 users, let's say for an example, then its an issue for UGS service to return all those users to us in a single go :) 
as you said it won't be a problem for smaller groups, but, can we ask the business or QA to find out whats the biggest group we have at the moment?  if not no issues, we can go ahead with these changes

Reply:
it would be nice if the UGS service provides an endpoint that returns an admin user for a given group, not sure if UGS lies within MDTP,

Reply:
Agreed.  Let's talk with the auth team today - although we have reduced the number of calls, we might end up with a huge JSON payload.  Not good.